File: RAI/Clase10/Python/Gato.py

class Gato:
    #Todos los metodos deben incorporar al self como parte de sus parametros
    #El constructor en python siempre se llama __init__()
    def __init__(self, name:str, sex:str, age:int, weight:int, color:str, texture:str):
        # En el constructor se definen e instancian los atributos del objeto
        self.__name = name
        self.__sex = sex
        self.__age = age
        self.__weight = weight
        self.__color = color
        self.__texture = texture
    
    def eat(self, comida:str):
        print(self.__name + " va a comer " + comida)

    def move(self):
        print(self.__name + " se va a mover...")

    def meow(self):
        print(self.__name + " va a hacer meowww...")

    def purr(self):
        print(self.__name + " se va a ronrronear...")
    
    def hunt_mice(self):
        print(self.__name + " se va a cazar ratones...")

    def get_name(self):
        return self.__name
    
    def get_color(self):
        return self.__color

    def set_color(self, nuevo_color:str):
        if nuevo_color is "blanco":
            self.__color = nuevo_color
        else:
            print("No es permitido el cambio de color")

File: RAI/Clase10/Java/Gato.java
public class Gato{
    //atributos
    //son variables
    private String name;
    private String sex;
    private int age;
    private int weight;
    private String color;
    private String texture;

    //metodos
    //son funciones
    //El primer metodo se llama constructor o builder
    //En java el constructor siempre tiene el nombre de la clase
    public Gato(String name, String sex, int age, int weight, String color, String texture){
        //this  hace referencia a la informacion propia del objeto
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    public void eat(String comida){
        System.out.println(this.name + " va a comer " + comida);
    }
    
    public void move(){
        System.out.println(this.name + " va a mover");
    }

    public void meow(){
        System.out.println(this.name + " hace meowww");
    }

    public void purr(){
        System.out.println(this.name + " va a ronrronear");
    }

    public void huntMice(){
        System.out.println(this.name + " va a cazar ratones");
    }

    public String getName(){
        return this.name;
    }

    public String getColor(){
        return this.color;
    }

    public void setColor(String nuevoColor){
        if(nuevoColor.equals("blanco"))
            this.color = nuevoColor;
        else
            System.out.println("No es posible cambiar el color");
    }


}